sale_layout_category_hide_detail/static/src/js/boolean_fa_icon_widget.js

```diff
@@ -36,28 +37,46 @@ odoo.define("sale_layout_category_hide_detail.boolean_fa_icon_widget", function
         // Private
         // --------------------------------------------------------------------------
 
+        _allowEdit: function () {
+            var allow = true;
+            if (
+                (this.model === "account.move.line" &&
+                    this.recordData.parent_state !== "draft") ||
+                (this.model === "sale.order.line" && this.recordData.state !== "draft")
+            ) {
+                allow = false;
+            }
+            return allow;
+        },
+
         /**
          * Render font-awesome icon based on state
          *
          * @override
          * @private
          */
+
         _render: function () {
-            // Set icon class
-            var fa_icons = this.attrs.options.fa_icons;
-            var icon_true = (fa_icons && fa_icons.icon_true) || "fa-check-square-o";
-            var icon_false = (fa_icons && fa_icons.icon_false) || "fa-square-o";
-            var fa_class = this.value ? icon_true : icon_false;
-            // Set tip message
-            var terminology = this.attrs.options.terminology;
-            var hover_true =
-                (terminology && _t(terminology.hover_true)) || _t("Click to uncheck");
-            var hover_false =
-                (terminology && _t(terminology.hover_false)) || _t("Click to check");
-            var tip = this.value ? hover_true : hover_false;
-            // Set template and add it to $el
-            var template = "<span class='fa %s' title='%s' aria-label='%s'></span>";
-            this.$el.empty().append(_.str.sprintf(template, fa_class, tip));
+            if (this.recordData.display_type === "line_section") {
```

Comment:
Why do you need this `if` if this is only applied for this widget?

Reply:
Actually it is iterating through all the lines (not only section lines), and for example at debug level it is a bit confusing; anyway I have removed it because in any case it would be something that should be implemented by odoo core.